packages/eslint-plugin/no-invalid-custom-properties/index.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2023, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ESLintUtils } from '@typescript-eslint/utils';
+
+/**
+ * TODO. For now this is to see whether the rule runs (it should error on all
+ * occurrences of the --cui-fg-success string).
```

Comment:
Should we limit this rule to only validate custom properties inside template literals?

Reply:
I thought about this, but there's more use cases:

```js
// JS
const COLOR_MAP = {
  default: "var(--cui-fg-normal)",
  success: "var(--cui-fg-success)",
}
```

```html
<!-- HTML -->
<p style="color:var(--cui-fg-success);">Success!</p>
```

```css
/* CSS */
.success {
  color: var(--cui-fg-success);
}
```

I think it would still make sense to cover these, especially since we're not sure how long we'll be using CSS-in-JS at the company.

Regex is fast so performance should not be affected, if that's your concern. I'll benchmark it anyways when I write the rule.

Reply:
Good point! Linting comments still feels a bit weird to me, but I suppose it's good to keep them up-to-date and valid as well 👍

Reply:
We can exclude comments though 🙂 I was thinking of trying this out. If I remember correctly, the ESLint rule context exposes raw text with and without comments

Reply:
Checked again—it seems that we can only grab tokens without comments when we use the AST. In the current implementation, we're getting raw lines of code from ESLint rule context, which I don't think even get parsed into an AST yet.

If the benchmark is really bad, I'll revisit the approach